src/node/node.c

```diff
@@ -356,16 +365,27 @@ static void set_value(struct s_plugin_conf *conf, const char *key,
 
 	if (dst_env == NULL) {
 		/* Allocate one */
-		if (conf->size == MAX_ENV_NB) {
+		if (conf->used == MAX_ENV_NB) {
 			fprintf(stderr, "ran out of internal env space\n");
 			abort();
 		}
+		if (conf->used == conf->size) {
+			conf->size = conf->size * 3 / 2;
```

Comment:
I would go the very naive road of doing some `realloc` every new insertion
Therefor relying on the fact that either : 

- `realloc` is doing some alignment magic anyway
- we don't have much items to add anyway, so the added effort isn't that big
- the code is much dumber therefore more bug proof.

Yet, your optimization isn't rocket science either, and already written, so let's keep it that way.